pkg/trait/knative.go

```diff
@@ -70,7 +71,8 @@ func (t *knativeTrait) Configure(e *Environment) (bool, *TraitCondition, error)
 	if e.Integration == nil {
 		return false, nil, nil
 	}
-	if !pointer.BoolDeref(t.Enabled, true) {
```

Comment:
Why do you need this?

Reply:
I noticed the `NewIntegrationConditionUserDisabled`  says the trait is explicitly disabled, so this sets the condition only when the user sets `enabled=false`.

Reply:
But I think that's exactly what `!pointer.BoolDeref(t.Enabled, true)` does. If the uses does not set explicitly to `false`, then it is always  `true`.